src/Storages/MergeTree/MergeTreeReadPool.h

```diff
@@ -254,8 +255,9 @@ class MergeTreeInOrderReadPoolParallelReplicas : private boost::noncopyable
             buffered_tasks.push_back({part.data_part->info, MarkRanges{}});
 
         extension.all_callback({
```

Comment:
Can you just add name of the class before initializer?

```suggestion
        extension.all_callback(InitialAllRangesAnnouncement{
```

you have to jump couple of times to get to the code which tells you which class is constructed.